File: 프로그래머스/lv0/120903. 배열의 유사도/배열의 유사도.java
class Solution {
    public int solution(String[] s1, String[] s2) {
        int result = 0;
        for(int i=0; i<s1.length; i++) {
            for(int j=0; j<s2.length; j++) {
                if( s1[i].equals(s2[j]) ) {
                    result++;
                }
            }
        }
        return result;
    }
}
/*
    접근법 :
    매개변수로 들어온 배열들을 반복문으로 접근해 해결해보았다.
    
    **주의 :
    처음에 if 문 안에 조건을 .equals 가 아닌 == 으로했는데 오류가났다.
    String 타입의 값을 비교하다보니 정확히 캐치를 못한 것 같다.

    == : 객체의 주소값 비교 -> 참조 타입 변수들 간의 연산은 동일한 객체를 참조하는지, 다른 객체를 참조하는지 알아볼 때 사용
    .equals() : 데이터값 비교 -> 참초 타입에서 더 정확
    
*/

File: 프로그래머스/lv0/120903. 배열의 유사도/배열의 유사도.js
function solution(s1, s2) {
    let result = s1.filter( (e) => { return s2.includes(e); });
    return result.length;
}
/*
    접근법 :
    1. s1의 요소와 s2의 요소를 비교해서 중복되는 요소의 길이를 구해보자
    2. filter() 안 콜백함수의 역할 -> s2의 요소와 비교해 중복되면 true 반환(요소 유지)
    3. includes() 를 사용해 s1의 요소가 s2에 들어있는지 검사 true/false 반환

    다른 풀이 :
    1. 위 코드보다는 번거롭지만 매개변수로 들어온 배열을 반복문으로 비교해 접근
    function solution(s1, s2) {
        let result = 0;
        for(let i=0; i<s1.length; i++) {
            for(let j=0; j<s2.length; j++) {
                if(s1[i] === s2[j]) {
                    result++;
                }
            }
        }
        return result;
    }
*/   